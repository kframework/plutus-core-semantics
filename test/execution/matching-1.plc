(program

(module Foo (import) (export () ())
  (define result
    (case (con Prelude.MkPair 3 7)
          (Prelude.MkPair (x y) (builtin multiplyInteger x y))))
  (define swapped
    (case (con Prelude.MkPair 1 2)
      (Prelude.MkPair (x y) (con Prelude.MkPair y x))))

)

)
