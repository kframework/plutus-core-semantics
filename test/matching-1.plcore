(program

(module Foo (imported) (exported () ()) (local () ())
  (define result
    (case (con SomeMod.Foo 1)
          (SomeMod.Foo (x)  (builtin addInteger 10 x))
          (SomeMod.NotThis ()  -1))))

)
