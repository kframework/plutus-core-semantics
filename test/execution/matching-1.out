<T> <k> . </k> <lenv> .Map </lenv> <genv> ( Foo . result ) |-> 11 </genv> <mod> Foo </mod> </T>
